Open Judge System/Services/OJS.Services/RedisCacheService.cs

```diff
@@ -178,9 +186,10 @@ public async Task RemoveAsync(string cacheId)
             }
             catch (RedisConnectionException ex)
             {
-                if (!this.MemoryCacheHasKey(ex.GetType().ToString(), ex.Message))
+                var exTypeAsString = ex.GetType().ToString();
+                if (!this.MemoryCacheHasKey(exTypeAsString, ex.Message))
                 {
-                    await this.emailSenderService.SendEmailAsync(this.devEmail, ex.GetType().ToString(), ex.Message);
+                    await this.emailSenderService.SendEmailAsync(this.devEmail, exTypeAsString, ex.Message);
```

Comment:
Extract this into a private method `HandleRedisConnectionException` and reuse it.

Reply:
Done